lib/vrl/compiler/src/value/kind.rs

```diff
@@ -25,8 +26,12 @@ pub use ::value::{
     Kind,
 };
 
-impl Value {
-    pub fn kind(&self) -> Kind {
+pub trait VrlValueKind {
+    fn kind_vrl(&self) -> Kind;
+}
+
+impl VrlValueKind for Value {
+    fn kind_vrl(&self) -> Kind {
         self.into()
     }
```

Comment:
If this change is needed because of the conflict with `pub fn kind(&self) -> &str` in `vector-core`, then I think it'd be better to rename that one to `kind_str`, since that one is (1) less often used, and (2) arguably badly named (ideally we'd get rid of that one, but we/I can do that in a follow-up PR).

Reply:
Sure, I assumed both of those were going away with the new `Kind` type so I didn't bother we the name too much. It's easy enough to rename them though